### PR TITLE
Upgrade Dockerfile to python3

### LINKED ISSUE #124: Make Dockerfile use Python 3
Python 2 is nearing EOL, weechat has switched to Python 3, and the README states that Python 3 is supported. The Dockerfile should use Python 3 dependencies.

### PR DESCRIPTION
**Pull request by nibalizer**
This fixes #124 

I have a different version [here](https://github.com/nibalizer/docker-images/blob/master/weechat-matrix/Dockerfile) which I think is cleaner but is a larger departure from the original file. Happy to do whatever.

### CODE REVIEW

**Comment by poljar** [comment]
I think we should go with the cleaner one, since we don't need to keep track of updating the deps twice there.

**Comment by nibalizer** [comment]
Ok I can swap it in!

**Comment by nibalizer** [comment]
This should also resolve #122

File: Dockerfile [hunk 1]
```diff
@@ -1,43 +1,45 @@
...
-    && cd weechat-matrix \
-    && make install
+FROM debian:testing-slim
+
+RUN echo 'debconf debconf/frontend select Noninteractive' | debconf-set-selections
+RUN apt-get update -y; apt-get install -q -y \
+  build-essential \
+  cmake \
+  git \
+  libolm-dev \
```
Comment (poljar) [inline Dockerfile]: We get libolm-dev from the repo, isn't libolm there as well? What version? We might skip the manual Olm build if it's new enough.

Comment (nibalizer) [reply Dockerfile]: Yes it looks like moving to testing gets us (libolm 3.1)[https://packages.debian.org/bullseye/libolm-dev] Let me play with this a bit and respin

**Comment by nibalizer** [comment]
Removing the build step also let me make the dockerfile way cleaner and remove all the build deps.

**Pull request merged**